esp-hal-common/Cargo.toml

```diff
@@ -81,11 +78,11 @@ serde      = { version = "1.0.190", features = ["derive"] }
 
 [features]
 esp32   = ["xtensa", "esp32/rt",   "procmacros/esp32",   "xtensa-lx/esp32",   "xtensa-lx-rt/esp32"]
-esp32c2 = ["riscv",  "esp32c2/rt", "procmacros/esp32c2", "portable-atomic?/unsafe-assume-single-core"]
-esp32c3 = ["riscv",  "esp32c3/rt", "procmacros/esp32c3", "portable-atomic?/unsafe-assume-single-core"]
+esp32c2 = ["riscv",  "esp32c2/rt", "procmacros/esp32c2", "portable-atomic/unsafe-assume-single-core"]
+esp32c3 = ["riscv",  "esp32c3/rt", "procmacros/esp32c3", "portable-atomic/unsafe-assume-single-core"]
 esp32c6 = ["riscv",  "esp32c6/rt", "procmacros/esp32c6"]
 esp32h2 = ["riscv",  "esp32h2/rt", "procmacros/esp32h2"]
-esp32s2 = ["xtensa", "esp32s2/rt", "procmacros/esp32s2", "xtensa-lx/esp32s2", "xtensa-lx-rt/esp32s2", "usb-otg", "portable-atomic?/unsafe-assume-single-core"]
+esp32s2 = ["xtensa", "esp32s2/rt", "procmacros/esp32s2", "xtensa-lx/esp32s2", "xtensa-lx-rt/esp32s2", "usb-otg", "portable-atomic/critical-section"]
```

Comment:
Apparently we shouldn't enable these in a library:

> critical-section
> ...
> Note:
> ...
> It is usually not recommended to always enable this feature in dependencies of the library.

Reply:
I read that too, however, I feel that although esp-hal is a library, its also not a platform-agnostic library, we know exactly what the intended target is capable of. If it causes problems in the future we can revert this.